mp/src/game/shared/momentum/run/mom_run_entity.cpp

```diff
@@ -67,7 +67,6 @@ void CMomRunEntity::OnZoneExit(CTriggerZone *pTrigger)
         break;
     }
 
-    pData->m_bMapFinished = false;
```

Comment:
Removing this negatively affects online/replay ghosts. Did you test playing back replays and restarting them? I would figure the panel would stay open for them with this change.

Perhaps moving this line to `OnZoneEnter`, specifically the `ZONE_TYPE_START` case would be good for all run entities. Then you can just keep the `SetLaggedMovement` call in the player and inheritance will set the `m_bMapFinished` bool to false for you.

Reply:
Removing it had no negative effect on replays since `m_bMapFinished` was set independently for replays, but I moved the common `m_bMapFinished = false` line to the base class anyway.

Reply:
Resolved in fcd31e2